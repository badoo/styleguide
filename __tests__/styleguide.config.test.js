import StyleguideConfig from '../styleguide.config';
import path from 'path';

describe('Config checks', () => {
    it('Config Structure', () => {
        expect(StyleguideConfig).toMatchSnapshot();
    });

    it('Config getExceptionForLoaders', () => {
        expect(StyleguideConfig.getExceptionForLoaders()).toMatchSnapshot();
    });

    it('Config getSections', () => {
        try {
            const getComponentsFromSections = (sections) => {
                const sectionList = sections.toString();
                const componentPaths = sectionList
                    .match(/require\((.+)\)/gm)
                    .map((path) => path.replace(/require\((.+)\)/g, '$1').replace(/'/g, ''));

                return Array.from(new Set(componentPaths));
            };

            const resolveComponentPathsFromComponentRoots = (components, getComponentRoots) => {
                const listOfResolvedComponents = getComponentRoots({ path }).map((root) =>
                    components.map((component) => path.parse(path.resolve(root, component)).name)
                );

                return listOfResolvedComponents.reduce(
                    (flatComponentsList, section) => flatComponentsList.concat(section),
                    []
                );
            };
            const components = StyleguideConfig.getSections
                ? getComponentsFromSections(StyleguideConfig.getSections)
                : undefined;
            const includePaths =
                components && StyleguideConfig.getComponentRoots
                    ? resolveComponentPathsFromComponentRoots(
                          components,
                          StyleguideConfig.getComponentRoots
                      )
                    : undefined;

            expect(includePaths).toMatchSnapshot();
        } catch (err) {
            console.log(err);
        }
    });

    it('Config getComponentRoots', () => {
        expect(
            StyleguideConfig.getComponentRoots({ path }).map((item) => path.parse(item).name)
        ).toMatchSnapshot();
    });

    it('Config getWebpackConfig', () => {
        expect(
            StyleguideConfig.getWebpackConfig({ path }).resolve.modules.map(
                (item) => path.parse(item).name
            )
        ).toMatchSnapshot();
    });
});
