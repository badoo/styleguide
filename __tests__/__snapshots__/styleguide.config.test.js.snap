// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config checks Config Structure 1`] = `
Object {
  "applyBabelToTypescriptCode": true,
  "getComponentRoots": [Function],
  "getExceptionForLoaders": [Function],
  "getSections": [Function],
  "getWebpackConfig": [Function],
  "hasResizableSandbox": false,
  "legendFontSize": 30,
  "placeholderFontSize": 30,
  "sandboxMaxWidth": undefined,
  "sandboxMinWidth": undefined,
}
`;

exports[`Config checks Config getComponentRoots 1`] = `
Array [
  "src",
]
`;

exports[`Config checks Config getExceptionForLoaders 1`] = `
Object {
  "jsLoader": Array [
    /spec\\\\\\.jsx\\$/,
  ],
}
`;

exports[`Config checks Config getSections 1`] = `
Array [
  "error-boundary",
  "component",
]
`;

exports[`Config checks Config getWebpackConfig 1`] = `
Array [
  "src",
  "node_modules",
]
`;
